File: Final-Lab/motifFinding.py
## Finding motif 
def subs(string1, string2):
    loc = []
    for i in range(len(string1)):
        if string2 == string1[i: i+len(string2)]:
            loc.append(i+1)
    return loc

if __name__ == "__main__":
    
    string1 = input()
    string2 = input()
    loc = subs(string1, string2)
    for i in loc:
        print(i, end=" ")

File: Final-Lab/motifFinding.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> subs(string string1, string string2) {
    vector<int> loc;
    for (int i = 0; i < string1.length(); i++) {
        if (string2 == string1.substr(i, string2.length())) {
            loc.push_back(i+1);
        }
    }
    return loc;
}

int main() {
    string string1, string2;
    cin >> string1 >> string2;
    vector<int> loc = subs(string1, string2);
    for (int i = 0; i < loc.size(); i++) {
        cout << loc[i] << " ";
    }
    return 0;
}
